pages/infinite-page/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useRef, useState } from "react";
+
+type Fn<T extends any[], R> = (...t: T) => R;
+
+const useDetectOnScreen = (
+  ref: React.RefObject<HTMLElement>,
+  onScreenHandler: Fn<[], void>,
+  offScreenHandler: Fn<[], void>
+) => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  useEffect(() => {
+    // initialize observer
+    const observer = new IntersectionObserver(
+      ([entry]: any) => {
+        setIsVisible(entry.isIntersecting);
```

Comment:
you can just call `onScreenHandler` and `offScreenHandler` inside here 

separating the `useEffect` twice may require 2 passes in the update, once to update `isVisible`, the other to call `[on|off]ScreenHander`